### PR TITLE
Operators for Point: +, -, +=, -=

### PR DESCRIPTION
**Pull request by vlolteanu**

### CODE REVIEW

File: Source/engine.h [hunk 1]
```diff
@@ -60,6 +60,26 @@ enum direction : uint8_t {
...
+	}
+
+	Point operator -(const Point &other)
+	{
+		return {x - other.x, y - other.y};
+	}
+
+	void operator +=(const Point &other)
+	{
+		*this = *this + other;
```
Comment (glebm) [inline Source/engine.h:76]: nit: Usually it's implemented the other way around (`+` in terms of `+=` and copy)

File: Source/engine.h [hunk 2]
```diff
@@ -60,6 +60,26 @@ enum direction : uint8_t {
...
+	{
+		return {x + other.x, y + other.y};
+	}
+
+	Point operator -(const Point &other)
+	{
+		return {x - other.x, y - other.y};
+	}
+
+	void operator +=(const Point &other)
```
Comment (glebm) [inline Source/engine.h:74]: Operators `+=` and `-=` should return `Point &` (`return *this`)

File: Source/engine.h [hunk 3]
```diff
@@ -60,6 +60,26 @@ enum direction : uint8_t {
 struct Point {
 	int x;
 	int y;
+
+	Point operator +(const Point &other)
```
Comment (glebm) [inline Source/engine.h:64]: From http://en.cppreference.com/w/cpp/language/operators:

> Binary operators are typically implemented as non-members to maintain symmetry

File: Source/engine.h (hunk 2 above)
Comment (vlolteanu) [reply Source/engine.h:74]: Yikes! Thanks for catching this.

File: Source/engine.h (hunk 3 above)
Comment (vlolteanu) [reply Source/engine.h:64]: It does make it look nicer, even though this struct is trivial.

File: Source/engine.h (hunk 1 above)
Comment (vlolteanu) [reply Source/engine.h:76]: I see no point doing it the other way around in this case. (Pun intended.)

Comment (glebm) [reply Source/engine.h:76]: There is no advantage to either approach in this case, so please use the common approach for consistency. There is no need for this one to be special.

File: Source/engine.h [hunk 4]
```diff
@@ -60,6 +60,28 @@ enum direction : uint8_t {
 struct Point {
 	int x;
 	int y;
+
+	friend Point operator +(const Point &a, const Point &b)
+	{
+		return {a.x + b.x, a.y + b.y};
+	}
```
Comment (glebm) [inline Source/engine.h:67]:
```suggestion
	friend Point operator +(Point a, const Point &b)
	{
		a += b;
		return a;
	}
```

etc

**Review by glebm — APPROVED** [review]

**Pull request merged**